system/jlib/jutil.cpp

```diff
@@ -2873,24 +2873,84 @@ jlib_decl StringBuffer &getTempFilePath(StringBuffer & target, const char * comp
     StringBuffer dir;
     if (pTree)
         getConfigurationDirectory(pTree->queryPropTree("Directories"),"temp",component,pTree->queryProp("@name"),dir);
-    if (!dir.length())
+    bool ok = false;
+    if (!dir.isEmpty())
     {
+        ok = recursiveCreateDirectory(dir.str());
+        if (ok)
+        {
 #ifdef _WIN32
-        char path[_MAX_PATH+1];
-        if(GetTempPath(sizeof(path),path))
-            dir.append(path).append("HPCCSystems\\hpcc-data");
-        else
-            dir.append("c:\\HPCCSystems\\hpcc-data\\temp");
+            return target.set(dir);
+#else
+            int srtn = access(dir.str(), R_OK|W_OK|X_OK);
```

Comment:
Yes.  There is a:
```
    if (checkDirExists(name))
```
in the recursiveCreateDirectory() code path, so that if it exists and its a dir then its ok, and if it is a file or any part of it is a file then a mkdir would fail and fall through to the next attempt.